system/repositories.rs

```diff
@@ -0,0 +1,112 @@
+pub mod user_repository {
+    use crate::concepts::{Credential, PasswordHash, User};
+    use crate::util::query_util::{execute_read_pipeline, execute_write_pipeline};
+    use answer::variable_value::VariableValue;
+    use concept::thing::thing_manager;
+    use concept::type_::type_manager::TypeManager;
+    use database::transaction::{TransactionRead, TransactionWrite};
+    use function::function_manager::FunctionManager;
+    use std::collections::HashMap;
+    use std::sync::Arc;
+    use storage::durability_client::WALClient;
+    use storage::snapshot::WriteSnapshot;
+    use thing_manager::ThingManager;
+    use typeql::parse_query;
+    use uuid::Uuid;
+    use crate::util::answer_util::get_string;
+
+    pub fn list(tx: TransactionRead<WALClient>) -> Vec<User> {
+        let query = parse_query(
+            "match (user: $u, password: $p) isa user-password; $u has name $n;"
+        ).unwrap();
+        let (tx, result) =
+            execute_read_pipeline(tx, &query.into_pipeline());
+        let rows = result.unwrap();
+        let mut users: Vec<User> = vec![];
+        for row in rows {
+            let name  = get_string(&tx, &row, "n");
+            users.push(User::new(name))
+        }
+        users
+    }
+
+    pub fn get(tx: TransactionRead<WALClient>, username: &str) -> Option<(User, Credential)> {
+        let query = parse_query(
+            format!(
+                "match
+                (user: $u, password: $p) isa user-password;
+                $u has name '{username}';
+                $p has hash $h, has salt $s;",
+                username = username
```

Comment:
Looks like you don't even need to pass an argument if you write `{username}`

Reply:
Fixed in 08f3441d77d54a2fc5d18b90a24b199c00f5aeac.